server/src/utils/auth.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { type AxiosError } from 'axios';
+
+export type UserInfo = {
+  email: string;
+  email_verified: string;
+  name: string;
+  nickname: string;
+  picture: string;
+  sub: string;
+  updated_at: string;
+};
+
+export type UserInfoResponse =
+  | {
+      success: true;
+      userInfo: UserInfo;
+    }
+  | {
+      success: false;
+      reason: 'AXIOS_ERROR';
+      error: AxiosError;
+    };
+
+export async function getUserInfo(
+  auth0Url: string,
+  auth0Token: string
+): Promise<UserInfoResponse> {
+  const userInfoUrl = `${auth0Url}/userinfo`;
+  const headers = { 'Authorization': `Bearer ${auth0Token}` };
+  const response = await axios.get(userInfoUrl, { headers })
+    .then((value) => ({ success: true as const, value }))
+    .catch((error) => ({ success: false as const, error }));
+
+  if (!response.success) {
+    return {
+      success: false,
+      reason: 'AXIOS_ERROR',
+      error: response.error,
```

Comment:
What is inside `error` object? This is coming from axios right? Can we extract only the information that is needed?

Reply:
To be honest, I looked at the Auth0 API docs and they're not very clear about the things we can expect from their API, error or success. The payloads they return are not clearly documented.

Reply:
I'm probably not looking in the right place, though.

Reply:
Can you try triggering an error and log it to see how it looks? We'll be able to see the error format that way and only grab the info we need, not the whole thing.

Reply:
I see you addressed this. Resolving.